extensions/amp-lightbox/0.1/amp-lightbox.js

```diff
@@ -309,6 +316,59 @@ class AmpLightbox extends AMP.BaseElement {
   getHistory_() {
     return Services.historyForDoc(this.getAmpDoc());
   }
+
+  /**
+   * Sets the document body to transparent to allow for frame "merging" if the
+   * element is under FIE.
+   * The module-level execution of setTransparentBody() only works on inabox,
+   * so we need to perform the check on element build time as well.
+   * @private
+   */
+  maybeSetTransparentBody_() {
+    if (this.getAmpDoc().win != this.win) { // in FIE
+      setTransparentBody(this.getAmpDoc().win, /** @type {!HTMLBodyElement} */ (
+          dev().assert(this.win.document.body)));
+    }
+  }
+}
+
+
+/**
+ * Sets the document body to transparent to allow for frame "merging".
+ * @param {!Window} win
+ * @param {!HTMLBodyElement} body
+ * @private
+ */
+function setTransparentBody(win, body) {
+  Services.vsyncFor(win).run({
+    measure(state) {
+      state.alreadyTransparent =
+          computedStyle(win, body)['background-color'] == 'rgba(0, 0, 0, 0)';
+    },
+    mutate(state) {
+      if (!state.alreadyTransparent && !getMode().test) {
+
+        // TODO(alanorozco): Create documentation page and link it here once the
+        // A4A lightbox experiment is turned on.
+        user().warn(TAG,
+            'The background of the <body> element has been forced to ' +
+            'transparent. If you need to set background, use an intermediate ' +
+            'container.');
+      }
+
+      // set as !important regardless to prevent changes
+      setImportantStyles(body, {background: 'transparent'});
+    },
+  }, {});
 }
 
+
+// TODO(alanorozco): refactor this somehow so we don't need to do a direct
+// getMode() check
+if (getMode().runtime == 'inabox') {
+  setTransparentBody(window, /** @type {!HTMLBodyElement} */ (
```

Comment:
can you move this into `maybeSetTransparentBody_()` we we treat inabox the same as FIE, which is our end goal.